src/kaggle.jl

```diff
@@ -27,7 +29,18 @@ function gen_kaggle_url(dataset)
     return "https://www.kaggle.com/api/v1/datasets/download/$dataset"
 end
 
-function kdownload(dataset, localdir)
+is_kaggle_url(url) = contains(url, KAGGLE_DOMAIN)
+
+function kaggle_url2dataset(url)
+    user_name = match(Regex("$KAGGLE_DOMAIN/([^\\/]+)/"), url).captures[]
+    dataset_name = match(Regex("$user_name/([^\\/]+)"), url).captures[]
```

Comment:
Why using two regex instead of one?

Reply:
the url shown as bellow has different pattern at the end:

```
"https://www.kaggle.com/ningjingyu/fetchtest"
"https://www.kaggle.com/ningjingyu/fetchtest/tasks"
```

any suggestion?

Reply:
What about optional regex? 
`r"$KAGGLE_DOMAIN/([^\\]+)/([^\\]+)?`

Reply:
`r_str` cannot handle string insertion, I'll use `Regex("$KAGGLE_DOMAIN/([^/]+)/([^/]+)")`